### PR TITLE
range loop refactoring

### PR DESCRIPTION
**Pull request by Joilnen**

### CODE REVIEW

File: OgreMain/src/OgreMeshSerializerImpl.cpp [hunk 1]
```diff
@@ -2962,9 +2900,9 @@ namespace Ogre {
         pushInnerChunk(stream);
         {
             // Get one set of detail per SubMesh
-            unsigned short numSubs, i;
-            numSubs = pMesh->getNumSubMeshes();
-            for (i = 0; i < numSubs; ++i)
+            // numSubs = pMesh->getNumSubMeshes();
```
Comment (paroj) [inline OgreMain/src/OgreMeshSerializerImpl.cpp:2903]: commented code

File: OgreMain/src/OgreMeshSerializerImpl.cpp [hunk 2]
```diff
@@ -2974,7 +2912,7 @@ namespace Ogre {
                         "MeshSerializerImpl::readMeshLodUsageGenerated");
                 }
 
-                SubMesh* sm = pMesh->getSubMesh(i);
+                // SubMesh* sm = pMesh->getSubMesh(i);
```
Comment (paroj) [inline OgreMain/src/OgreMeshSerializerImpl.cpp:2915]: commented code

File: OgreMain/src/OgreMeshSerializerImpl.cpp [hunk 3]
```diff
@@ -604,11 +589,11 @@ namespace Ogre {
         // Bone assignments
         if (!pSub->mBoneAssignments.empty())
         {
-            SubMesh::VertexBoneAssignmentList::const_iterator vi;
-            for (vi = pSub->mBoneAssignments.begin();
-                 vi != pSub->mBoneAssignments.end(); ++vi)
+            auto bass = pSub->mBoneAssignments.size();
+            while (bass)
```
Comment (paroj) [inline OgreMain/src/OgreMeshSerializerImpl.cpp:593]: why not for?

Comment (paroj) [reply OgreMain/src/OgreMeshSerializerImpl.cpp:593]: @Joilnen

Comment (paroj) [reply OgreMain/src/OgreMeshSerializerImpl.cpp:593]: this one is still outstanding

**Review by paroj — CHANGES_REQUESTED** [review]
the while loop should be change to for

**Comment by Joilnen** [comment]
backing to normal *for*, *range for* rise not used variable compile errors in some builds

**Comment by Joilnen** [comment]
> the while loop should be change to for

did

**Pull request merged**